File: src/main/java/cn/booling/bakahdt/Command.java
package cn.booling.bakahdt;

import net.mamoe.mirai.event.events.MessageEvent;

import java.util.HashMap;
import java.util.Map;
import java.util.function.Consumer;

public class Command {
    public static Map<String, Command> commands = new HashMap<>();

    public String name;
    public String info;
    public Consumer<MessageEvent> consumer;

    private Command(String name, String info, Consumer<MessageEvent> consumer) {
        this.name = name;
        this.info = info;
        this.consumer = consumer;
    }

    public static void registerCommand(String name, String info, Consumer<MessageEvent> consumer) {
        Command cmd = new Command(name, info, consumer);
        commands.put(name, cmd);
        TextFields.HELP = TextFields.HELP.plus("&" + name + "  " + info + "\n");
    }
}

File: src/main/kotlin/cn/booling/bakahdt/command/Command.kt
package cn.booling.bakahdt.command

import cn.booling.bakahdt.*
import kotlinx.serialization.*
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.*
import java.io.*

var simpleCommands = mutableMapOf<String, SimpleCommand>()

@Serializable
data class SimpleCommand(
    val name: String,
    val info: String? = null,
    val permission: String = "member",
    val messages: MutableList<String> = mutableListOf()
)

@Serializable
data class CommandMap(var commands: List<SimpleCommand> = mutableListOf())

fun loadSimpleCommands() {
    val commandMap = Json.decodeFromString<CommandMap>(File("commands.json").readText())
    commandMap.commands.forEach { cmd ->
        if (cmd.messages.isNotEmpty()) simpleCommands[cmd.name] = cmd
        if (cmd.info != null) TextFields.HELP = TextFields.HELP.plus("\n$IDENTIFIER${cmd.name}  ${cmd.info}")
        logger.info("Loaded command: ${cmd.name} ${cmd.info}")
    }
    logger.info("Simple Commands Reloaded")
}

fun List<String>.toMessage(): String {
    var message = ""
    this.forEach {
        message += "$it\n"
    }
    message = message.trimEnd('\n')
    return message
}
